#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

gen = ParameterGenerator ()
gen.add("resolution", double_t, 0, "resolution of voxel grid", 0.1, 0.0, 1.0)
gen.add("publish_marker", bool_t, 0, "publish marker", True)
marker_color_enum = gen.enum([gen.const("flat", str_t, "flat", "flat"),
                              gen.const("z", str_t, "z", "z"),
                              gen.const("x", str_t, "x", "x"),
                              gen.const("y", str_t, "y", "y")],
                             "color")
gen.add("marker_color", str_t, 0, "marker color", "z", edit_method=marker_color_enum)
gen.add("marker_color_alpha", double_t, 0, "", 0.8, 0.0, 1.0)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "OctreeVoxelGrid"))
